masu/external/downloader/local/local_report_downloader.py

```diff
@@ -56,15 +57,52 @@ def __init__(self, customer_name, auth_credential, bucket, report_name=None, **k
         self.customer_name = customer_name.replace(' ', '_')
 
         LOG.debug('Connecting to local service provider...')
+        prefix, name = self._extract_names(bucket)
 
         if report_name:
             self.report_name = report_name
         else:
-            self.report_name = os.listdir(bucket)[0]
-        self.base_path = bucket
+            self.report_name = name
+        self.report_prefix = prefix
+
+        LOG.info('Found report name: %s, report prefix: %s', self.report_name, self.report_prefix)
+        if self.report_prefix:
+            self.base_path = '{}/{}/'.format(bucket, self.report_prefix)
+        else:
+            self.base_path = bucket
+        self.bucket_path = bucket
         self.bucket = bucket.replace('/', '_')
         self.credential = auth_credential
 
+    def _extract_names(self, bucket):
+        """
+        Find the report name and prefix given the bucket path.
+
+        Args:
+            bucket (String): Path to the local file
+
+        Returns:
+            (String, String) report_prefix, report_name
+
+        """
+        daterange = '\d{8}-\d{8}'
+        full_path = ''
+        for item in os.walk(bucket, followlinks=True):
+            if not item[2]:
+                if any(re.findall(daterange, date) for date in item[1]):
+                    full_path = item[0]
+                    break
+        directories = full_path[len(bucket):]
+        parts = directories.strip('/').split('/')
+
+        report_name = None
+        report_prefix = None
+        if len(parts) == 2:
```

Comment:
This block can be simplified and made clearer, like this:

```
report_name = parts.pop(0)
report_prefix = parts.pop(0) if parts else None
```

Reply:
We would still need to figure out the length of `parts` since if there's a prefix provided, the `report_name` would get the prefix value since it comes second in the list.  So the only difference would be getting the values out directly (like the current implementation) or popping them off the end.  Am I missing something?

Reply:
oh on second look, removing the 0 it would probably work.

Reply:
Ah. Yeah. Small bug in my example. :sweat_smile:

Reply:
might be a little more too it to, but standby